cpp/src/arrow/util/endian.h

```diff
@@ -124,12 +127,23 @@ static inline T ToBigEndian(T value) {
   return ByteSwap(value);
 }
 
+template <typename T, size_t N>
+static inline std::array<T, N> ToBigEndian(std::array<T, N> array) {
```

Comment:
This API is a bit ambiguous. Should it also byteswap the `T`s?

Reply:
Removed this API with general name.